Dockerfile

```diff
@@ -25,7 +28,24 @@ RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build \
 
 WORKDIR /kargo/bin
 
+####################################################################################################
+# ui-builder
+####################################################################################################
+FROM --platform=$BUILDPLATFORM docker.io/library/node:18.16.1 AS ui-builder
```

Comment:
Are we 100% sure we can get away with `--platform=$BUILDPLATFORM` here?

This could introduce a problem IF any of the packages used build native extensions.

I admittedly don't know enough about the packages used by the UI to say whether or not that is something we're going to run into.

cc @rbreeze @rpelczar

Reply:
Sorry this was a bit of a mindless copy and paste from a technique we use in another Dockerfile. It may not be applicable now, but I do suspect we may need this for multi-arch builds down the road. I will revert this for now.

Reply:
Actually, removing `--platform=$BUILDPLATFORM` increased the image build time from 6 minutes to 12 minutes so I think we want this. I tested that the image that I built on darwin/arm64 worked on linux/amd64. Also, the fact that we use this technique in another project successfully gives me confidence.

Reply:
Cool. I agree this is the way to go as long as we've verified this doesn't break any native extensions that might have been in the dependency graph.